source/base/tensor_product_polynomials.cc

```diff
@@ -38,35 +38,38 @@ namespace internal
     compute_tensor_index(const unsigned int,
                          const unsigned int,
                          const unsigned int,
-                         unsigned int (&)[dim])
+                         std::array<unsigned int, dim> &)
     {
       Assert(false, ExcNotImplemented());
     }
 
+    template <>
```

Comment:
Ha, interesting! It turns out that we declare a general template further up and that these functions you're modifying here are specializations. But that's a bit unnecessary -- they could just be overloads, which turns out this function here was before you added the `template <>`. Just remove the `template<>` from all three functions and make them overloads, rather than specializations. Then you can also remove the `<1>` again.